icu4c/source/tools/upropdump/upropdump.cpp

```diff
@@ -0,0 +1,178 @@
+// © 2016 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include <iostream>
+#include "toolutil.h"
+#include "uoptions.h"
+#include "cmemory.h"
+#include "charstr.h"
+#include "unicode/uchar.h"
+#include "unicode/errorcode.h"
+#include "unicode/uniset.h"
+#include "unicode/putil.h"
+#include "unicode/umutablecptrie.h"
+#include "writesrc.h"
+
+U_NAMESPACE_USE
+
+/*
+ * Global - verbosity
+ */
+UBool VERBOSE = FALSE;
+UBool QUIET = FALSE;
+
+UBool haveCopyright = TRUE;
+UCPTrieType trieType = UCPTRIE_TYPE_SMALL;
```

Comment:
Are you building "small" tries to reduce the "dump" file size? If you were to use these tries for runtime *lookups*, then "fast" ones would be better.

Reply:
Made it into a command-line flag.

13e28462bbed57b95ef20be3af1b5ed474d4b174